src/main/java/org/bigredbands/mb/models/Point.java

```diff
@@ -23,6 +23,26 @@ public Point interpolate(Point b, float t) {
         return new Point(this.x*(1-t) + b.x*t, this.y*(1-t) + b.y*t);
     }
 
+    public Point normalize() {
+        return this.multiply(1.0f / (float) this.distance(0, 0));
+    }
+
+    public Point orthogonal() {
```

Comment:
Can we rename this to something like `getOrthogonalVector` or something similar that clarifies this is meant to be used when a point represents a vector?

If this represents a 2D point (as it does when representing the start and end of a RankPosition), orthogonal doesn't make sense.

Reply:
I modeled these changes after the [`javafx.geometry.Point2D`](https://docs.oracle.com/javase/8/javafx/api/javafx/geometry/Point2D.html) class, which represents a point _or_ vector (as opposed to the AWT [`Point2D`](https://docs.oracle.com/javase%2F8%2Fdocs%2Fapi%2F%2F/java/awt/geom/Point2D.html), which only represents a point). The JavaFX one doesn't have an equivalent `orthogonal()` method, but it does have `normalize()` + a bunch of other operations that only apply to vectors (`dotProduct()`, `crossProduct()`, etc.).

So while I'd like to keep the names (to introduce slightly less churn to eventually cut over completely to JavaFX, although that's a huge refactor in and of itself), I can make the "point or vector" purpose of `Point` clearer with class & function documentation.

Reply:
That works for me.  As a case study, I was confused reading them for a while for how a 2D point could have these operations until I got further into the code, so I think having the additional clarity would help.